### PR TITLE
Unify `File_Format_Metadata` with `File_For_Read`

### LINKED ISSUE #8555: Unify handling of file format detection
Recently #8403 introduced `File_Format_Metadata` that is a bit similar (but different) to `File_For_Read`.

We want to unify these types - ensuring that most fields are optional and ensuring that our file format detectors can work with that.

Along these lines, we may want to unify `for_web` and `for_file_read` as well - essentially currently `for_file_read` relies on `File_For_Read`, whereas `for_web` relies on a subset of `File_Format_Metadata` - once we merge these types, we could have `for_file_read` handle also the content-type case from `for_web` - as I'm not sure if we really need distinction if the thing is coming from web (do we? cc: @jdunkerley) - we mostly just want to be able to detect the format based on available information - whether this is file name extension, `Content-Type` (or other mime-type info), or possibility to read first bytes from the file.

---

We may want to add `File_Format_Metadata` to the `read` function as well, similarly as it is added to `read_stream`. The presence of additional metadata may be useful to further enhance the file format information that may be used for decoding the file.

The current instance of that use case is - when the user specifies the `Excel` type explicitly (which defaults `xls_format=Infer`, once we actually `read` or `read_stream` it - we want to change the `Infer` to actually be a proper `True|False` value and the metadata allow us to do so.

Currently, we default `Plain_Text`'s `encoding` to `utf_8`. We could similarly default it to `Infer` for reading and when the user requests to decode a request as `Plain_Text`, we could rely on the metadata to further enhance the format with the encoding based on the headers (still defaulting to UTF-8 if data is not available, of course).

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

- Closes #8555
- Refactors the file format detection logic, compacting lots of repetitive logic for HTTP handling into helper functions.
- Some updates to CODEOWNERS.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by radeusgd** [comment]
A few minor things I was planning to do before merging this:
- [x] change the default `encoding` from `utf_8` to `Infer` for `(Response|Response_Body).decode_as_text` (and `decode_as_json`), to have better logic: if `Infer`, if the `Content-Type` contains an encoding specified - use that, if not found, still fallback to `utf_8` as currently.
  - Currently, even if the headers specify another encoding, we still use UTF-8 by default, unless the user actively changes it. That is bad - if the headers give us more detailed information - we should be relying on that.
- [x] Add tests for this encoding auto-detection.
- [x] Add a test for detecting the filename from `Content-Disposition` and URI path.
  - Probably add endpoints in `http-test-helper` that return no `Content-Type` but one of them returns `Content-Disposition` - and ensure that the format (e.g. `.tsv`) is still correctly detected.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -34,6 +34,7 @@ Cargo.toml
 /project/ @4e6 @jaroslavtulach @hubertp
 /test/ @jdunkerley @radeusgd @GregoryTravis
 /tools/ @4e6 @jaroslavtulach @radeusgd
+/tools/http-test-helper/ @radeusgd @jdunkerley @GregoryTravis
```
Comment (radeusgd) [inline .github/CODEOWNERS]: I think it makes sense to hand over reviewing the http-test-helper that is solely used for Enso tests to the libraries team, no need to bother engine with it 🙂

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/XML/XML_Format.enso [hunk 2]
```diff
@@ -1,25 +1,27 @@
...
-    for_file_read : File_For_Read -> XML_Format | Nothing
-    for_file_read file:File_For_Read =
-        case file.extension of
-            ".xml" -> XML_Format
-            _ -> Nothing
+    for_read : File_Format_Metadata -> XML_Format | Nothing
+    for_read file:File_Format_Metadata =
+        content_type = file.interpret_content_type
+        from_content_type = content_type.if_not_nothing <|
+            if content_type.base_type.ends_with "/xml" then XML_Format else Nothing
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/XML/XML_Format.enso]: I think we should be more explicit here and do `application/xml` and `text/xml`.
We can also match ending with `+xml` as well.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Network/HTTP/Response.enso [hunk 3]
```diff
@@ -162,3 +173,28 @@ type Response
         type_pair = ["type", "Response"]
         cons_pair = ["constructor", "Value"]
         JS_Object.from_pairs [type_pair, cons_pair, ["headers", self.headers], ["body", self.body], ["code", self.code]]
+
+    ## PRIVATE
+    get_header : Text -> Text | Nothing
+    get_header self name:Text =
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Network/HTTP/Response.enso]: Can we follow the `get` style here and have a `(~default=Nothing)`?
Allows us to put defaults in or error then as needed.

File: distribution/lib/Standard/Base/0.0.0-dev/src/System/File_Format.enso [hunk 4]
```diff
@@ -117,73 +112,70 @@ type File_Format
...
     ## A file format for plain text files with the specified encoding.
+
+       Arguments:
+       - encoding: The encoding to use.
+         If the encoding is set to `Infer`, it will be inferred from metadata if
+         available, falling back to UTF-8. For writing, `Infer` will also imply
+         the UTF-8 encoding as the default.
     @encoding Encoding.default_widget
-    Plain_Text (encoding:Encoding=Encoding.utf_8)
+    Plain_Text (encoding:Encoding|Infer = Encoding.utf_8)
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/System/File_Format.enso]: Shouldn't we default to `Infer` here?

**Review by jdunkerley — APPROVED** [review]
A few little tweaks please

File: distribution/lib/Standard/Base/0.0.0-dev/src/System/File_Format.enso (hunk 4 above)
Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/System/File_Format.enso]: Sure, I was thinking about it, but didn't want to change the default unconsulted.

I'll amend it.

**Review by GregoryTravis — APPROVED** [review]

**Review by JaroslavTulach — APPROVED** [review]

**Pull request merged**